src/predictors/VectorNonlinearOracle.jl

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2024: Triad National Security, LLC
+# Copyright (c) 2024: Oscar Dowson and contributors
+#
+# Use of this source code is governed by a BSD-style license that can be found
+# in the LICENSE.md file.
+
+"""
+    VectorNonlinearOracle(x)
+
+A wrapper struct for creating an `Ipopt.VectorNonlinearOracle`.
+
+!!! warning
+    To use [`VectorNonlinearOracle`](@ref), your code must load the `Ipopt`
+    package.
+    ```julia
+    import Ipopt
+    ```
+"""
+struct VectorNonlinearOracle{P} <: AbstractPredictor
+    predictor::P
+    device::String
+    hessian::Bool
+
+    function VectorNonlinearOracle(
+        predictor::P;
+        device::String = "cpu",
+        hessian::Bool = true,
+    ) where {P}
+        return new{P}(predictor, device, hessian)
+    end
+end
+
+function build_vector_nonlinear_oracle end
+
+function MathOptAI.add_predictor(
+    model::JuMP.AbstractModel,
+    predictor::VectorNonlinearOracle,
+    x::Vector;
+    kwargs...,
+)
+    set = build_vector_nonlinear_oracle(predictor, length(x))
+    y = JuMP.@variable(model, [1:set.output_dimension])
+    con = JuMP.@constraint(model, [x; y] in set)
+    return y, Formulation(predictor, y, [con])
+end
```

Comment:
What happens if `x` contains numbers or expressions instead of just variables? Should we raise an error or add the intermediate variables automatically?

Reply:
JuMP will automatically reformulate by adding additional variables and an equality constraint